src/mem/freelist.h

```diff
@@ -740,6 +740,25 @@ namespace snmalloc
         UNUSED(domesticate);
 #endif
       }
+
+      /**
+       * Returns length of free list if you call close.
+       *
+       * The Builder is not always emptied when you call close,
+       * this returns the length of the list if you had just called
+       * close.
+       */
+      [[nodiscard]] size_t length_after_close() const
```

Comment:
```suggestion
      [[nodiscard]] size_t length_available() const
```

Reply:
I wasn't happy with the name.  I was trying to make something that made sense for both `RANDOM` and `!RANDOM`.  I have changed to using SFINAE, so it only exists for `RANDOM` and I think it is clearer.

Reply:
And RANDOM is synonymous with defined(SNMALLOC_CHECK_CLIENT)?  Yes, if somewhat circuitously: the `Metaslab` `free_queue` field is a `freelist::builder` whose `RANDOM` template parameter is derived from `SNMALLOC_CHECK_CLIENT`.  Sounds good.

Reply:
RANDOM is one feature that is enabled with SNMALLOC_CHECK_CLIENT.  At some point I would like to pull them apart, but now yet ;-)